packages/cli/src/ui/App.tsx

```diff
@@ -4,1190 +4,181 @@
-    }
-    return getAllGeminiMdFilenames();
+    return fromSettings
+      ? Array.isArray(fromSettings)
+        ? fromSettings
+        : [fromSettings]
+      : getAllGeminiMdFilenames();
   }, [settings.merged.contextFileName]);
 
   const initialPrompt = useMemo(() => config.getQuestion(), [config]);
+  const initialPromptSubmitted = useRef(false);
   const geminiClient = config.getGeminiClient();
 
   useEffect(() => {
     if (
       initialPrompt &&
       !initialPromptSubmitted.current &&
-      !isAuthenticating &&
-      !isAuthDialogOpen &&
-      !isThemeDialogOpen &&
-      !isEditorDialogOpen &&
-      !showPrivacyNotice &&
+      !uiState.isAuthenticating &&
+      !uiState.isAuthDialogOpen &&
+      !uiState.isThemeDialogOpen &&
+      !uiState.isEditorDialogOpen &&
+      !uiState.showPrivacyNotice &&
       geminiClient?.isInitialized?.()
     ) {
-      submitQuery(initialPrompt);
+      uiActions.handleFinalSubmit(initialPrompt);
       initialPromptSubmitted.current = true;
     }
   }, [
     initialPrompt,
-    submitQuery,
-    isAuthenticating,
-    isAuthDialogOpen,
-    isThemeDialogOpen,
-    isEditorDialogOpen,
-    showPrivacyNotice,
+    uiActions,
+    uiState.isAuthenticating,
+    uiState.isAuthDialogOpen,
+    uiState.isThemeDialogOpen,
+    uiState.isEditorDialogOpen,
+    uiState.showPrivacyNotice,
     geminiClient,
   ]);
 
-  if (quittingMessages) {
+  const errorCount = useMemo(
+    () =>
+      uiState.filteredConsoleMessages
+        .filter((msg) => msg.type === 'error')
+        .reduce((total, msg) => total + msg.count, 0),
+    [uiState.filteredConsoleMessages],
+  );
+
+  if (uiState.quittingMessages) {
     return (
       <Box flexDirection="column" marginBottom={1}>
-        {quittingMessages.map((item) => (
+        {uiState.quittingMessages.map((item) => (
           <HistoryItemDisplay
             key={item.id}
             availableTerminalHeight={
-              constrainHeight ? availableTerminalHeight : undefined
+              uiState.constrainHeight ? availableTerminalHeight : undefined
             }
             terminalWidth={terminalWidth}
             item={item}
             isPending={false}
-            config={config}
           />
         ))}
       </Box>
     );
   }
 
   const mainAreaWidth = Math.floor(terminalWidth * 0.9);
-  const debugConsoleMaxHeight = Math.floor(Math.max(terminalHeight * 0.2, 5));
-  // Arbitrary threshold to ensure that items in the static area are large
-  // enough but not too large to make the terminal hard to use.
-  const staticAreaMaxItemHeight = Math.max(terminalHeight * 4, 100);
-  const placeholder = vimModeEnabled
-    ? "  Press 'i' for INSERT mode and 'Esc' for NORMAL mode."
-    : '  Type your message or @path/to/file';
+  const nightly = version.includes('nightly');
+  const pendingHistoryItems = [
+    ...uiState.pendingSlashCommandHistoryItems,
+    ...uiState.pendingGeminiHistoryItems,
+  ];
+
+  const dialogsVisible =
```

Comment:
make this a helper method and separate all the dialog specific state a seperate interface under uiState just for the dialogState.